app/assets/stylesheets/govuk_publishing_components/components/_input.scss

```diff
@@ -16,3 +16,24 @@
     padding-left: govuk-spacing(6);
   }
 }
+
+.gem-c-input__wrapper {
+  display: flex;
+}
+
+.gem-c-input:focus {
+  z-index: 1; // increase z-index on focus so it appears over the suffix
+}
+
+.gem-c-input__suffix {
+  // scss-lint:disable PlaceholderInExtend
+  // sass-lint:disable placeholder-in-extend
+  @extend .govuk-input;
```

Comment:
I've only briefly played with this but it seems like instead of doing this `@extend` the `govuk-input` class could be added to this element. Would save a few lines of code.

(I point this out partly because I have a vague memory of `@extend` causing us headaches at some point to do with compilation, but I can't remember where unfortunately)

Reply:
No real headaches (aside from generating quite a bit of css that may be too much for this scenario), but not ideal and sanctioned by linter when used on classes (not placeholders). 

The main reason we went with `@extend` is that we want to make sure the suffix styles are aligned with the input styles in case the later change. There is a bit of logic around the height of the input which is generated differently based on `$govuk-typography-use-rem` flag and I wanted to avoid it. However, I removed the `@extend` and got the logic in – we can propose the Design System team to expose `.govuk-input` styles to be used as a placeholder if there will be future needs around it.